src/target/target.cc

```diff
@@ -373,6 +373,15 @@ Target::Target(const Map<String, ObjectRef>& config) {
   data_ = std::move(target);
 }
 
+Target::Target(Target target, Target host) {
+  ObjectPtr<TargetNode> n = make_object<TargetNode>(*target.get());
+  CHECK(!n->host.defined())
+      << "ValueError: Adding a host to a target whose host field has been defined";
```

Comment:
Please correct me if my question doesn't make sense, but am I correct to assume that this check will always pass, as you're adding the field directly to the macro ` TVM_REGISTER_TARGET_KIND`, which is used by all existing targets, and therefore `host` is a field for all of them?

Reply:
Hi, I created a new unit tests to demonstrate the ValueError.